### DISCUSSION TITLE
Sugestões de temas para TCC

### DISCUSSION THREAD

**Discussion by felipecesr** [frontendbr/forum#524] [Discussão]
Olá pessoal, estou terminando minha Pós-Graduação em Análise, Projeto e Gerência de Sistemas e queria falar sobre JavaScript no TCC, não tenho um tema definido e queria a ajuda de vocês com sugestões.

**Comment by vitor-mendrone-deel** [comment]
Faala, @felipecesr, editei o assunto da sua pergunta, ok? Assim ficamos com um tópico que pode servir de referência pra galera no futuro.

**Comment by vitor-mendrone-deel** [comment]
Alias, uma sugestão de tema legal: programação assíncrona (simultaneidade)

O Concurrency model de JS é bem diferente de outras linguagens do "mainstream" tipo Java, acaba sendo um assunto extenso e complexo

**Comment by kivervinicius** [comment]
O que acha falar sobre realtime, apesar de ser um assunto "antigo" tem muitas áreas que podem/tiram proveito disto

**Comment by mahenrique94** [comment]
@felipecesr concordo com o @kivervinicius, um tema legal seria o **realtime**, recentemente precisei implementa-lo e tive grandes dificuldades para faze-lo.

**Comment by gabrieldarezzo** [comment]
Vale NodeJs? ou só no mundo do front?
Especifica isso tmb se possível

**Comment by felipecesr** [comment]
Ficou melhor assim @Mendrone. Pode ser NodeJS @gabrieldarezzo.

**Comment by kivervinicius** [comment]
Você deverá montar seu tcc de qual forma? Projeto, Escrito?

**Comment by felipefialho** [comment]
@lucasjs @nayarahilton :)

**Comment by NathanReuter** [comment]
Podes falar sobre a diferença de paradigmas, usando JS no aspecto de programação funcional VS programação orientada a objetos

**Comment by felipecesr** [comment]
Em forma de artigo @kivervinicius

**Comment by lucasjs** [comment]
@felipecesr só escrever um artigo científico?

**Comment by felipecesr** [comment]
sim @lucasjs

**Comment by lucasjs** [comment]
@felipecesr Ufa! Se fosse um projeto, aí seria outros 500. Bom, eu particularmente falaria sobre os  493280482390482938904023 frameworks. A evolução, surgimento, pontos positivos e negativos, influência no mercado de trabalho, como isso influi nas pessoas entrando na área, etc.

**Comment by expalmer** [comment]
Fala sobre a Libuv

**Comment by yanmagale** [comment]
Então @felipecesr , penso que seu assunto, além do gosto e da vontade por pesquisa-lo, muito dependerá também do seu tempo. Quando mais recente for o assunto, maior será o tempo para buscar as fontes certas(se não tiver bom no inglês, levará mais tempo para ler os artigos), sem falar na própria escrita do assunto em si. Digo isso pois acabei de me formar em ciências da computação e meu tcc foi sobre arquitetura flux, comparando com o MVC, gastei em tornos de 3 a 4 mêses somente na pesquisa, juntando todos os documentos e lendo bastante. Como quase não existiam documentos em PT-BR, levei um bom tempo procurando e lendo os materiais em encontrava.

Quanto as sugestões: Programação Funcional com JS, algum framework JS que você queria aprender melhor, evoluções da linguagem ... a lista realmente é grande.

Meu conselho seria falar sobre algo que você realmente queira se aprofundar e aprender melhor.

**Comment by mensonones** [comment]
Estou fazendo o meu TCC sobre Frameworks JS para front-end. Fazer uma análise comparativa .

**Comment by Moniquespl** [comment]
PARA QUEM COMO EU NÃO SABE POR ONDE COMEÇAR O TCC KK TENHO UM CURSO EM CONTA E MARAVILHOSO PARA INDICAR CHAMADO TCC SEM DRAMA, ME AJUDOU MTO ESPERO QUE AJUDE VCS TBM =)
https://go.hotmart.com/O5942588N

**Comment by wadson-pereira** [comment]
@mahenrique94 , o que seria "realtime"?

**Comment by Diego-Marchesi** [comment]
estou fazendo pós  em POO com java, e está difícil encontrar um tema que agrade meu orientador na Estácio, se alguém tiver algum tema para sugerir agradeço.
Att.

**Comment by thiagofrancisquete** [comment]
E quanto a temas de tg pratico? 

To querendo fazer algo relacionado ao front, mas artigo eu acho muito entediante. E na minha faculdade tem muito TG com análise de dados publicos com python ultimamente, praticamente todo mundo fazendo a mesma coisa, não queria ser mais um... mas é complicado tirar uma idéia pratica do nada, é proibido fazer sistema de gerenciamento de academia, mercadinho e afins, que era a antiga _"moda"_ antes da analise de dados

**Comment by expalmer** [comment]
Procura conversar com teus professores, pergunta para eles se tem algum tema, ou trabalho incompleto que seja interessante. Compra a idéia e faz o TCC em cima disso. As vezes tem gente que tem ideias fantásticas porém não tem skill.

**Comment by qrozrafa** [comment]
Olá pessoal, estou terminando minha graduação em tecnologo de sistema para internet e, gostaria de realizar o projeto do tcc em react.js. Não tenho um tema definido, gostaria da ajuda de vocês.

obs: 
- Preciso criar um artigo cientifico (introdução, metedologia, referencia, conclusao) sobre o tema;
- Criar um projeto com os metodo  HTTP (front-end e back-end)

**Reply by felquis** [reply]
O que exatamente vc precisa de ajuda?

Gosto de usar o https://remix.run para front e back.

**Reply by qrozrafa** [reply]
A encontrar um tema kk queria algo simples e funcional

**Reply by felquis** [reply]
Sugestões do ChatGPT
- Desenvolvimento de um sistema de gerenciamento de projetos para equipes remotas.
- Criação de um sistema de e-commerce personalizado para pequenas empresas.
- Implementação de segurança em um sistema de gerenciamento de banco de dados.
- Desenvolvimento de um aplicativo mobile para gerenciamento de tarefas.
- Criação de um sistema de gerenciamento de conteúdo para websites.

**Reply by gabrieldarezzo** [reply]
Pelo que olhei no seu perfil, vc já tem habilidades de js.
Nesse caso minha sugestão seria algo em Node + Front em react.

Sobre a ideia.
Uma caderno de recados online.

Usando:
- Offline First.
Recebeu uma nova mensagem?
O recebedor está online?
Se sim sincronizar apenas aquela mensagem no storage local.

- PushNotification
Opa mensagem chegou, dispara o push.
Ainda acho uma tecnologia maneira, mesmo 99% das vezes ser algum site chato que entrei apenas 1x na vida, usar push em apps pode ser uma boa.

- Semântica e acessibilidade


O restante é otimização, cacheado oq der, reduzir o tamanho do payload otimizar o Node/Back pra responder rapidinho, fazer um front magrinho só com o necessário.
